Neos.ContentRepository.Core/Classes/ContentGraphFinder.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core;
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\ContentGraphInterface;
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamDoesNotExistYet;
+use Neos\ContentRepository\Core\SharedModel\Exception\WorkspaceDoesNotExist;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+
+/**
+ * A finder for ContentGraphInterface bound to contentStream/Workspace
+ *
+ * @see ContentGraphInterface
+ * @api only for read access during write operations and in services
+ */
+final class ContentGraphFinder implements ContentGraphFinderInterface
```

Comment:
Maybe rather `ContentGraphProvider` ... i understand the relation to the `WorkspaceFinder` name wise but there are no `find*` methods.

Reply:
See comment below but totally not married to the name here, lets discuss what we find best and I change it.